_assets/javascripts/lib/set-redirect-url.js

```diff
@@ -1,10 +1,21 @@
-window.onload = function () {
-  var exclusions = [
-    '/media/hubspot-subscribe-form-embed'
-  ];
+window.onload = function() {
+  var exclusions = ["/media/hubspot-subscribe-form-embed"];
   if (exclusions.indexOf(window.location.pathname) === -1) {
-    var isRedirectUrlSet = new Event('redirect-url-set');
-    document.cookie = "redirectUrl=" + encodeURIComponent(window.location.href) + ";domain=.crossroads.net;path=/";
+    var isRedirectUrlSet = new Event("redirect-url-set");
+    var cookieName = "redirectUrl=";
+    var domainValue = document.domain;
+    var cookieValue = document.cookie
+      .split(";")
+      .find(selectedCookie => selectedCookie.startsWith(" " + cookieName));
+    var isCorrectName = cookieValue ? cookieValue.includes(domainValue) : false;
+
+    if (isCorrectName == true || cookieValue == undefined) {
```

Comment:
Just so I'm understanding correctly... you are evaluating `cookieValue == undefined` so as to not overwrite a previously set value?